src/safeposix/syscalls/fs_calls.rs

```diff
@@ -544,7 +545,8 @@ impl Cage {
     //------------------------------------FSTATFS SYSCALL------------------------------------
 
     pub fn fstatfs_syscall(&self, fd: i32, databuf: &mut FSData) -> i32 {
-        let unlocked_fd = self.filedescriptortable[fd as usize].read();
+        let tmp = self.get_filedescriptor(fd).unwrap();
```

Comment:
i think we should have a better name than tmp for these, can you not call this plus the read() unlock on the same line?

Reply:
`self.get_filedescriptor(fd).unwrap().read()` will cause lifetime error, so I separate them

Reply:
ok lets change all the tmp's to 'checkedfd'